### PR TITLE
spanner-jdbc: Step 26 - JDBC Statement and PreparedStatement implementation

### PR DESCRIPTION
**Pull request by olavloite**
Adds implementations for java.sql.Statement and java.sql.PreparedStatements. The most important task for these classes is to convert positional JDBC parameters to Cloud Spanner named parameters.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+  }
+
+  @Override
+  public void setNull(int parameterIndex, int sqlType) throws SQLException {
+    checkClosed();
+    parameters.setParameter(parameterIndex, null, sqlType, null);
+  }
+
+  @Override
+  public void setBoolean(int parameterIndex, boolean x) throws SQLException {
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java]: minor - please rename all `x` variables - `value` maybe?

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+  }
+
+  @Override
+  public void setArray(int parameterIndex, Array x) throws SQLException {
+    checkClosed();
+    parameters.setParameter(parameterIndex, x, Types.ARRAY);
+  }
+
+  @Override
+  public ResultSetMetaData getMetaData() throws SQLException {
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java]: test case?

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcStatement.java [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            String.format("Update count too large for int: %d", updateCounts[index]),
+            Code.OUT_OF_RANGE);
+      }
+      res[index] = (int) updateCounts[index];
+    }
+    return res;
+  }
+
+  @Override
+  public ResultSet getGeneratedKeys() throws SQLException {
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcStatement.java]: test case

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java (hunk 1 above)
Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java]: `x` variables have been renamed to `value`.

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java (hunk 2 above)
Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java]: Added test case.

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcStatement.java (hunk 3 above)
Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcStatement.java]: Added test case.

**Pull request merged**